.github/workflows/integration-test.yaml

```diff
@@ -104,68 +92,109 @@ jobs:
-      # TODO: Ditto for /home/runner/.pnpm-store/v3
-      # TODO: Ditto for /home/runner/.nuget/packages
-      # TODO: Ditto for /var/cache/apt/archives/
-      - name: Playwright setup
-        working-directory: frontend
-        if: ${{ inputs.run-playwright }}
-        run: pnpm install && pnpm pretest
-      - name: Set up Playwright dependencies
-        working-directory: frontend
-        if: ${{ inputs.run-playwright && inputs.runs-on == 'self-hosted' }}
-        run: sudo pnpm exec playwright install-deps
       - name: Build for tests
         run: "dotnet build /p:MercurialVersion=$MERCURIAL_VERSION"
         env:
           MERCURIAL_VERSION: ${{ inputs.hg-version }}
 
-      - name: Integration tests (Playwright)
-        if: ${{ inputs.run-playwright }}
+      - name: Integration tests (.NET)
+        if: ${{ matrix.test == 'dotnet' }}
         env:
           TEST_SERVER_HOSTNAME: ${{ vars.TEST_SERVER_HOSTNAME }}
           #      this is not a typo, we need to use the lf domain because it has a cert that hg will validate
           TEST_STANDARD_HG_HOSTNAME: ${{ vars.TEST_STANDARD_HG_HOSTNAME }}
           TEST_RESUMABLE_HG_HOSTNAME: ${{ vars.TEST_RESUMABLE_HG_HOSTNAME }}
           TEST_PROJECT_CODE: 'sena-3'
           TEST_DEFAULT_PASSWORD: ${{ secrets.TEST_USER_PASSWORD }}
-        working-directory: frontend
-        run: pnpm test
+          # 1.5gb max heap size, must be in hex https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector#heap-limit
+          # the self hosted runner only has 2gb so this should prevent it from running out of memory
+          DOTNET_GCHeapHardLimit: '0x2CB417800'
 
-      - name: Integration tests (.NET)
+        run: dotnet test --output ./bin --logger trx --results-directory ./test-results --filter Category=Integration
+      - name: Publish unit test results
+        uses: EnricoMi/publish-unit-test-result-action/composite@8885e273a4343cd7b48eaa72428dea0c3067ea98 # v2.14.0
+        if: ${{ always() && !env.act }}
+        with:
+          check_name: Integration Tests ${{ inputs.runs-on }} for Mercurial ${{ inputs.hg-version }}
+          files: ./test-results/*.trx
+      - name: Publish results to testspace
         if: always()
+        run: testspace "[.Net Integration/$${{ inputs.runs-on }} HG ${{ inputs.hg-version }}]./test-results/*.trx"
+
+  playwright-test:
+    timeout-minutes: 120
+    name: Playwright tests on ${{ inputs.runs-on }} for Mercurial ${{ inputs.hg-version }} on ${{ inputs.environment }}
+    permissions:
+      checks: write
+    environment:
+      name: ${{ inputs.environment }}
+    runs-on: ${{ inputs.runs-on }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: testspace-com/setup-testspace@v1.0.6
+        with:
+          domain: ${{ github.repository_owner }}
+      - name: Setup self-hosted dependencies
+        if: ${{ inputs.run-playwright && inputs.runs-on == 'self-hosted' }}
```

Comment:
I think we can [move `inputs.run-playwright` to the job](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif) so we don't need it on so many steps.

Reply:
I'd forgotten that jobs can have an `if` condition. That's way better. Fixed in commit c31294d7.